test/sql/hnsw_dist_func.sql

```diff
@@ -0,0 +1,133 @@
+CREATE TABLE small_world_l2 (
+    id varchar(3),
+    vector real[]
+);
+
+CREATE TABLE small_world_cos (
+    id varchar(3),
+    vector real[]
+);
+
+CREATE TABLE small_world_ham (
+    id varchar(3),
+    vector integer[]
+);
+
+CREATE TABLE small_world_l2_vec (
+    id varchar(3),
+    vector vector(3)
```

Comment:
Would be good to vactor this into a separate file (e.g. `pgvector_compat.sql`)
We will need to then move all the other tests that require pgvector into this file so we could make pgvector dependency optional for running tests as well.